frappe/desk/doctype/bulk_update/bulk_update.py

```diff
@@ -71,3 +73,63 @@ def show_progress(docnames, message, i, description):
 			title = message,
 			description = description
 		)
+
+@frappe.whitelist()
+def get_contact(name, doctype):
+	out = frappe._dict()
+
+	if doctype == "Purchase Order":
```

Comment:
Framework code should not be agnostic of doctype in other apps (in this case ERPNext)

You can pull contact_fields from hooks that will be in app repository.